core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -87,15 +87,24 @@ object StaticFile {
             else `Transfer-Encoding`(TransferCoding.chunked)
           val headers = Headers(lenHeader :: lastModHeader ::: contentType)
 
-          Some(
-            Response(
-              headers = headers,
-              body = readInputStream[F](F.delay(url.openStream), DefaultBufferSize, blocker)
-            ))
-        } else {
-          urlConn.getInputStream.close()
-          Some(Response(NotModified))
-        }
+          F.delay(url.openStream)
```

Comment:
I think it makes sense to delay `url.openStream` (which is apparently a blocking operation) in the `blocker` as well as `urlConn.getInputStream.close` below, doesn't it?

Reply:
true